supertokens_python/recipe/thirdparty/providers/github.py

```diff
@@ -71,4 +76,33 @@ def Github(input: ProviderInput) -> Provider:  # pylint: disable=redefined-built
     if input.config.token_endpoint is None:
         input.config.token_endpoint = "https://github.com/login/oauth/access_token"
 
+    if input.config.validate_access_token is None:
+        input.config.validate_access_token = validate_access_token
+
     return NewProvider(input, GithubImpl)
+
+
+async def validate_access_token(
+    access_token: str, config: ProviderConfigForClient, _: Dict[str, Any]
+):
+    client_secret = "" if config.client_secret is None else config.client_secret
+    basic_auth_token = base64.b64encode(
+        f"{config.client_id}:{client_secret}".encode()
+    ).decode()
+
+    url = f"https://api.github.com/applications/{config.client_id}/token"
+    headers = {
+        "Authorization": f"Basic {basic_auth_token}",
+        "Content-Type": "application/json",
+    }
+    payload = json.dumps({"access_token": access_token})
+
+    resp = requests.post(url, headers=headers, data=payload)
```

Comment:
do not directly use requests.. see how we are making API calls. For examlpe, we are using utility functions like `do_get_request`, which is used in this file. 

There is also `do_post_request` from that file which you should use.

Reply:
Done